ssz/src/main/java/org/ethereum/beacon/ssz/creator/ConstructorObjCreator.java

```diff
@@ -39,29 +38,30 @@
    * @return Pair[success or not, created instance if success or null otherwise]
    */
   @Override
-  public <C> Pair<Boolean, C> createObject(Class<? extends C> clazz,
+  public <C> C createObject(Class<? extends C> clazz,
       List<Pair<SSZField, Object>> fieldValuePairs) {
     Class[] params = new Class[fieldValuePairs.size()];
     for (int i = 0; i < fieldValuePairs.size(); i++) {
-      Pair<SSZSchemeBuilder.SSZScheme.SSZField, Object> pair = fieldValuePairs.get(i);
-      SSZSchemeBuilder.SSZScheme.SSZField field = pair.getValue0();
-      switch (field.multipleType) {
-        case LIST:
-          {
-            params[i] = List.class;
-            break;
-          }
-        case ARRAY:
-          {
-            params[i] = Array.newInstance(field.type, 0).getClass();
-            break;
-          }
-        default:
-          {
-            params[i] = field.type;
-            break;
-          }
-      }
+      Pair<SSZField, Object> pair = fieldValuePairs.get(i);
+      SSZField field = pair.getValue0();
+      params[i] = field.getRawClass();
+//      switch (field.multipleType) {
```

Comment:
Why do you need to keep this? It looks like it's already in `createInstanceBuilder`'s

Reply:
`createInstanceBuilder` uses `ObjCreator`
Removed commented code